### PR TITLE
Improve evaluation load speed

### PR DESCRIPTION
**Pull request by SepsiLaszlo**

### CODE REVIEW

File: app/services/evaluation_point_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class EvaluationPointCalculator
+  attr_reader :users, :user_count
+
+  def initialize(users)
+    @users      = users
+    @user_count = users.length.to_f
```
[the remark this answers is not in the document]
Comment (SepsiLaszlo) [reply app/services/evaluation_point_calculator.rb]: I used it for calcualting the averages but I fogot to remove it when implemented, that only eligible users are counted in average.

File: app/decorators/evaluation_user_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+class EvaluationUserDecorator < UserDecorator
+  attr_accessor :point_request, :point_details, :entry_request
+
+  def set_evaluation(evaluation)
+    @point_request = user.point_requests.find { |pr| pr.evaluation_id == evaluation.id }
+    @point_details = @point_request.point_details
+    @entry_request = user.entry_requests.find { |er| er.evaluation_id == evaluation.id }
+  end
```
[the remark this answers is not in the document]
Comment (SepsiLaszlo) [reply app/decorators/evaluation_user_decorator.rb]: The second option I think is more straight forward. I updated the decorator according to it!

File: app/controllers/judgements_controller.rb [hunk 3]
```diff
@@ -16,11 +16,14 @@ def show
...
-                        .map(&:user)
-                        .sort_by(&:full_name)
-    @users = UserDecorator.decorate_collection @users
+    @users = User.joins(point_requests: :evaluation)
+                 .where('evaluations.id': params[:evaluation_id])
+                 .where.not('point_requests.point': 0)
+                 .includes(:entry_requests, point_requests: [point_details: :principle])
+                 .sort_by(&:full_name)
+    @users = EvaluationUserDecorator.decorate_collection @users
+    @users.each { |user| user.set_evaluation(@evaluation) }
```
[the remark this answers is not in the document]
Comment (SepsiLaszlo) [reply app/controllers/judgements_controller.rb]: I did not know about the additional context argument. It is weird that it is not in the gem's README.
This makes using the decorator more pleasant, thank you!

[the remark this answers is not in the document]
Comment (falusi94) [reply app/controllers/judgements_controller.rb]: Yes, usually it's worth to check either the [rdoc](https://rdoc.info/github/drapergem/draper/Draper/Decorator#decorate_collection-class_method) or the source code.

**Review by falusi94 — APPROVED** [review]
🔥

**Pull request merged**